src/wallet/wallet.cpp

```diff
@@ -569,7 +574,7 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         Unlock(strWalletPassphrase);
 
         // if we are using HD, replace the HD seed with a new one
-        if (auto spk_man = m_spk_man.get()) {
+        if (auto spk_man = GetLegacyScriptPubKeyMan()) {
```

Comment:
Are we not going to replace the seed for descriptor wallets?

Reply:
No, descriptor wallets have a different way of doing it.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17261#discussion_r363792482

> No, descriptor wallets have a different way of doing it.

In commit "Box the wallet: Add multiple keyman maps and loops" (c729afd0a3b74a3943e4c359270beaf3e6ff8a7b):

It seems bad to do nothing at all and keep using unencrypted seeds for non-legacy wallets, and the comment here doesn't here doesn't explain any legacy/nonlegacy distinction. It would seem safer to trigger some kind of error for non-legacy wallets if planned behavior hasn't been implemented yet, than to just do nothing.

If there is supposed to be legacy / non-legacy distinction, it would seem better to move legacy code out of CWallet into LegacyScriptPubKeyMan in a `PostEncrypt` or similar ScriptPubKeyMan virtual method